src/main/java/com/profiq/codexor/GenerateDocumentationAction.java

```diff
@@ -63,20 +67,43 @@ public void actionPerformed(@NotNull AnActionEvent e) {
             ProgressManager.getInstance().run(new Task.Modal(e.getProject(), "Waiting for OpenAI", true) {
                 @Override
                 public void run(@NotNull ProgressIndicator indicator) {
-                    try {
-                        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
+                    var responseFuture = client.sendAsync(request, HttpResponse.BodyHandlers.ofString());
+
+                    responseFuture.thenAccept(response -> {
                         if (response.statusCode() == 200) {
                             var responseParsed = (new Gson()).fromJson(response.body(), Response.class);
                             var responseText = responseParsed.getChoices()[0].getMessage().getContent();
                             String docstring = parseDocstring(responseText);
                             var app = ApplicationManager.getApplication();
                             app.invokeLater(() -> showEditor(e, docstring));
                         } else {
+                            indicator.cancel();
                             showError(response.body());
                         }
-                    } catch (IOException | InterruptedException exception) {
-                        showError("Can't reach OpenAI at the moment");
+                    });
+
+                    responseFuture.exceptionally(e -> {
+                        if (e instanceof IOException || e instanceof InterruptedException) {
+                            showError("Can't reach OpenAI at the moment");
+                        }
+                        return null;
+                    });
+
+
+                    while (!responseFuture.isDone()) {
```

Comment:
I wonder if there is some other way of checking whether the user clicked the Cancel button than a while loop. But it works.